### ISSUE TITLE
when using fooocus-mre as api server and switching base model, NotImplementedError: Cannot copy out of meta tensor; no data!Exception in thread Thread-2 (worker) thrown

### ISSUE THREAD

**Issue by lanyusan** [MoonRide303/Fooocus-MRE#83]
I was evaluating Fooocus-MRE as an api backend. I wrapped the asyc worker to a fastapi endpoint, without launching gradio server and frontend. 

All went well until I tried the use case of switching base model. The following error was thrown.

It was fine with Fooocus-MRE app when switching base model through gradio frontend.

I have not idea why. I did some search. It seems this error happens with other AI projects. But I couldn't find a fix by myself.

If this can be addressed, Fooocus-MRE can be used as a stable backend.

```
NotImplementedError: Cannot copy out of meta tensor; no data!Exception in thread Thread-2 (worker):
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/modules/async_worker.py", line 379, in worker
    handler(task)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/modules/async_worker.py", line 289, in handler
    imgs = pipeline.process_diffusion(task['c'], task['uc'], steps, switch, width, height, task['task_seed'],
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/modules/default_pipeline.py", line 324, in process_diffusion
    decoded_latent = core.decode_vae(vae=xl_base_patched.vae, latent_image=sampled_latent)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/modules/core.py", line 87, in decode_vae
    return opVAEDecode.decode(samples=latent_image, vae=vae)[0]
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/repositories/ComfyUI-from-StabilityAI-Official/nodes.py", line 267, in decode
    return (vae.decode(samples["samples"]), )
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/repositories/ComfyUI-from-StabilityAI-Official/comfy/sd.py", line 195, in decode
    model_management.free_memory(memory_used, self.device)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/repositories/ComfyUI-from-StabilityAI-Official/comfy/model_management.py", line 330, in free_memory
    m.model_unload()
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/repositories/ComfyUI-from-StabilityAI-Official/comfy/model_management.py", line 301, in model_unload
    self.model.unpatch_model(self.model.offload_device)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/repositories/ComfyUI-from-StabilityAI-Official/comfy/model_patcher.py", line 269, in unpatch_model
    self.model.to(device_to)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1145, in to
    return self._apply(convert)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 797, in _apply
    module._apply(fn)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 797, in _apply
    module._apply(fn)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 797, in _apply
    module._apply(fn)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 820, in _apply
    param_applied = fn(param)
  File "/home/fqye/projects/Fooocus/Fooocus-MRE/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1143, in convert
    return t.to(device, dtype if t.is_floating_point() or t.is_complex() else None, non_blocking)
NotImplementedError: Cannot copy out of meta tensor; no data!  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()

```

**Comment by MoonRide303** [comment]
@lanyusan I would need some kind of reproduction steps. Could you provide simplest possible version of what you're trying to do, that would allow me to reproduce the problem on my own machine?

**Comment by lanyusan** [comment]
@MoonRide303 

I have created a repo at https://github.com/lanyusan/Fooocus-MRE

However when I tried to create a minimum version of api server, Fooocus seems not able to use gpu for inference.

It is really strange. I originally used runpod's handler wrapper to run the api, which was able to run Fooocus MRE with gpu. 

I guess your webui / gradio or runpod's wrapper has setup cuda environment appropriately.

I couldn't figure out the problem. I am not much skilled in cuda/ai/nvidia drivers etc.

I am wondering if yo can help take a look. The CPU version is really slow. I couldn't really run the test.

It is probably too much to ask. This isn't super urgent for me. No need to spend time on it if you are busy.

Thanks. 

Please:

1. clone the repo.
2. check out branch 83
3. create virtual environment
4. run python api.py at root folder
5. in another winow, run `node test.mjs`

**Comment by lanyusan** [comment]
@MoonRide303 

Hello,

I have spent some time to build a reproducible repo.
Please follow the following steps:

1. clone `https://github.com/lanyusan/Fooocus-MRE`
2. checkout branch 83
3. create and activate virtual environment
4. run `pip install -r requirements_versions.txt`
5. run `pip install -r api_requirements.txt`
6. launch the api server from a terminal with `python rp_handler.py  --rp_serve_api`
7. open another counsole, run the test script: `node api_test.mjs`
8. observer the api server terminal. You will notice that when base mode is switched back after 2nd generate, the exception: `NotImplementedError: Cannot copy out of meta tensor; no data!` is thrown

You need to place a second base model under the `models/checkpoints` folder.

I use this model: https://civitai.com/models/123307/sdvn7-nijistylexl in the test script `api_test.mjs`.

You need to modify the base model name in the api_test.mjs script:

```
let count = 2; //1000;

// change the base model name to what is placed under models/checkpoints folder

const base_models = [
  "sd_xl_base_1.0_0.9vae.safetensors",
  "sdvn7Nijistylexl_v1.safetensors",
];

for (let i = 0; i < count; i++)
  for (const modelFile of base_models) {
    await fetch("http://localhost:8000/runsync", {
      method: "POST",
      headers: {
        "content-type": "application/json",
      },
      body: JSON.stringify({
        input: {
          base_model_name: modelFile,
          prompt: "a puppy",
        },
      }),
    });
  }

```

Thanks for this great repo. I hope this problem can be fixed so that i can use it as our back end with multiple base model support in house.

**Comment by lanyusan** [comment]
@MoonRide303 
Sorry for bugging you. Would you please take a look at this? 
Thanks.

**Comment by MoonRide303** [comment]
@lanyusan I tried to launch it today - I needed to remove specific version from runpod to make to conflicts between requirements files (it ended up with 0.9.9). After running step 6 I started generating spam on console, whole screens full of messages like that:
```
WARN   | RUNPOD_WEBHOOK_GET_JOB not set, switching to get_local
WARN   | test_input.json not found, skipping local testing
ERROR  | Error while getting job: 'NoneType' object has no attribute 'get'
INFO   | No job available, waiting for the next one.
WARN   | RUNPOD_WEBHOOK_GET_JOB not set, switching to get_local
WARN   | test_input.json not found, skipping local testing
ERROR  | Error while getting job: 'NoneType' object has no attribute 'get'
INFO   | No job available, waiting for the next one.
WARN   | RUNPOD_WEBHOOK_GET_JOB not set, switching to get_local
WARN   | test_input.json not found, skipping local testing
ERROR  | Error while getting job: 'NoneType' object has no attribute 'get'
INFO   | No job available, waiting for the next one.
```

And node test (step 7) wasn't able to connect with it:
```
(base) PS E:\Fooocus\Fooocus-MRE-api> node .\api_test.mjs
node:internal/deps/undici/undici:11576
    Error.captureStackTrace(err, this);
          ^

TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11576:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///E:/Fooocus/Fooocus-MRE-api/api_test.mjs:10:5 {
  cause: Error: connect ECONNREFUSED ::1:8000
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16) {
    errno: -4078,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '::1',
    port: 8000
  }
}
```

**Comment by MoonRide303** [comment]
@lanyusan You can try updating to v2.0.78.5 - I've just removed meta tensor usage, it might help.

**Comment by lanyusan** [comment]
@MoonRide303 
Runpod 0.9.x doesn't support local api server mode. 1.x.x does have some dependency conflicts but could be ignored

I will try the new fix tomorrow when back in office. Don't have a GPU powered PC at home. 

Thanks again for the quick response.

**Comment by lanyusan** [comment]
@MoonRide303 

The new version doesn't throw the exception and can switch base models back and forth.

Thanks again.

**Comment by MoonRide303** [comment]
@lanyusan Good to hear - closing as solved in v2.0.78.5, then.
